src/mvt_sql.py

```diff
@@ -0,0 +1,455 @@
+# -*- coding: utf-8 -*-
+import sys
+import math
+from mapcss import _test_feature_compatibility
+
+reload(sys)
+sys.setdefaultencoding("utf-8")  # a hack to support UTF-8
+
+mapped_cols = {}
+osm2pgsql_avail_keys = {}
+
+
+def escape_sql_column(name, type, asname=False):
+    if name in mapped_cols:
+        if asname:
+            return mapped_cols[name] + ' as "%s"' % name
+        else:
+            return mapped_cols[name]
+
+    name = name.strip().strip('"')
+    type = {"line": "way", "area": "way"}.get(type, type)
+    if type in osm2pgsql_avail_keys.get(name, ()) or not osm2pgsql_avail_keys:
+        return '"' + name + '"'
+    elif not asname:
+        return "(tags->'" + name + "')"
+    else:
+        return "(tags->'" + name + "') as \"" + name + '"'
+
+
+def pixel_size_at_zoom(z, l=1):
+    """
+    Converts l pixels on tiles into length on zoom z
+    """
+    return int(math.ceil(l * 20037508.342789244 / 512 * 2 / (2 ** z)))
+
+
+def get_sql(condition, obj):
+    # params = [re.escape(x) for x in self.params]
+    params = condition.params
+    t = condition.type
+    if t == "eq":  # don't compare tags against sublayers
+        if params[0][:2] == "::":
+            return ("", "")
+    try:
+        column_name = escape_sql_column(params[0], type=obj)
+
+        if t == "eq":
+            return params[0], "%s = '%s'" % (column_name, params[1])
+        if t == "ne":
+            return (
+                params[0],
+                "(%s != '%s' or %s IS NULL)" % (column_name, params[1], column_name),
```

Comment:
```suggestion
                "(%s IS DISTINCT FROM '%s')" % (column_name, params[1]),
```

Reply:
Same in https://github.com/kothic/kothic/blob/9ff0aafe4a64404487dc5d4e373a5d46a082cb34/src/mapcss/Condition.py#L117